torch/csrc/distributed/c10d/c10d_frontend.cpp

```diff
@@ -0,0 +1,133 @@
+#include <torch/csrc/distributed/c10d/c10d_frontend.h>
+
+#include <sstream>
+#include <stdexcept>
+
+namespace c10d {
+
+int64_t DistributedC10d::getRank(const std::shared_ptr<ProcessGroup>& group) {
+  if (rankNotInGroup(group)) {
+    return -1;
+  }
+
+  // Note: This assumes we replace group.WORLD with default_pg_. Otherwise,
+  // we need another conditional to check whether group is WORLD and then use
+  // default_pg_ explicitly.
+  return group->getRank();
+}
+
+int64_t DistributedC10d::getWorldSize(const std::shared_ptr<ProcessGroup>& group) {
+  if (rankNotInGroup(group)) {
+    return -1;
+  }
+
+  // Note: This assumes we replace group.WORLD with default_pg_. Otherwise,
+  // we need another conditional to check whether group is WORLD and then use
+  // default_pg_ explicitly.
+  return group->getRank();
+}
+
+int64_t DistributedC10d::getGroupSize(
+    const std::shared_ptr<ProcessGroup>& group) const {
+  // Note: This assumes we replace group.WORLD with default_pg_. Otherwise,
+  // we need another conditional to check whether group is WORLD and then use
+  // default_pg_ explicitly.
+  if (group == default_pg_) {
+    default_pg_->getSize();
+  }
+
+  auto it = pg_group_ranks_.find(group);
+  if (it == pg_group_ranks_.end()) {
+    throw std::runtime_error("The given group does not exist");
+  }
+
+  return it->second.size();
+}
+
+std::shared_ptr<ProcessGroup> DistributedC10d::worldProcessGroup() {
+  checkDefaultPg();
+  return default_pg_;
+}
+
+bool DistributedC10d::rankNotInGroup(
+    const std::shared_ptr<ProcessGroup>& group) const {
+  // Note: This assumes we replace group.WORLD with default_pg_. Otherwise,
+  // we need another conditional to check whether group is WORLD and then use
+  // default_pg_ explicitly.
+  if (group == default_pg_) {
+    return false;
+  }
+  return group == nullptr;
+}
+
+int64_t DistributedC10d::getGroupRank(
+    const std::shared_ptr<ProcessGroup>& group,
+    const int64_t rank) const {
+  // Note: This assumes we replace group.WORLD with default_pg_. Otherwise,
```

Comment:
probably add the note to the header instead, to reduce the dups of the comment.

Reply:
Done.